docs: add guides to cloud nodes creation and usage (#1548)
ockam-network/ockam: documentation/guides/draft/rust/get-started/xx-discovery-using-cloud-node/README.md

# Discovery using cloud node

## Introduction

In the previous guide we learned how to [create a cloud node](../xx-cloud-node) and connect to it using the TCP transport.

> Cloud nodes can be used as proxies to connect devices not exposed by public hostnames or IPs. In this case the cloud node would redirect the traffic from one node to another using the transport connections established by devices.

Reviewer: We should wordsmith this with something similar to the NAT statement elsewhere. A place to highlight network challenges

## Discovery using forwarding

In this guide we're going to use the "forwarding_service" - a Hub Node service which allows a device to connect to another device using a temporary route alias.

In order to do that, a device we call "responder" is going to create an alias first, which can be used by another device we call "initiator" to send messages to responder.

> After discovery step both devices would have a route to each other, which they can use to exchange messages.

Reviewer: ```suggestion
After discovery both devices will have routes to each other, which they can use to exchange messages.
```

> <img src="./Discovery.png" width="70%">

Reviewer: First place we introduce the terminology `alias`. I think this is a nice succinct word better than "forwarding address"

## Application code

### Responder

First create a responder at:

```
touch examples/xx-discovery-using-cloud-node-responder.rs
```

Add the following code to this file:

```rust
use ockam::{Context, RemoteForwarder, Result, TcpTransport};
use ockam_get_started::Echoer;

#[ockam::node]
async fn main(ctx: Context) -> Result<()> {
    // Create a cloud node by going to https://hub.ockam.network
    let cloud_node_tcp_address = "<Your node host copied from hub.ockam.network>:4000";

    // Initialize the TCP Transport.
    let tcp = TcpTransport::create(&ctx).await?;

    // Create a TCP connection to your cloud node.
    tcp.connect(cloud_node_tcp_address).await?;

    // Create an echoer worker
    ctx.start_worker("echoer", Echoer).await?;

    let forwarder = RemoteForwarder::create(&ctx, cloud_node_tcp_address, "echoer").await?;
    println!(
        "Forwarding address of echoer: {}",
        forwarder.remote_address()
    );

    Ok(())
}
```

This node will run an `echoer` worker to reply to the messages sent from the initiator node.

### Run responder

> You need to get the temporary forwarding address from the Hub Node in order to configure the initiator properly.

Reviewer: I keep coming back to the word "temporary" here. It's accurate, but I wonder if we can standardize it. It is almost session-ish. It is valid for the lifecycle of the registering service, yeah?

Author: Maybe we should use the word "alias" here. Although I was trying to highlight that it's generated.
